base-api-client/src/main/java/com/cloudesire/platform/apiclient/baseclient/AbstractCloudesireClient.java

```diff
@@ -23,13 +24,26 @@
     protected ObjectMapper mapper;
 
     protected AbstractCloudesireClient( ObjectMapper mapper, String baseUrl, String userAgent )
+    {
+        this(
+                mapper,
+                baseUrl,
+                userAgent,
+                new TimeoutBuilder()
+                        .setConnectTimeout( 30, TimeUnit.SECONDS )
+                        .setReadTimeout( 60, TimeUnit.SECONDS )
+                        .build()
+        );
+    }
+
+    protected AbstractCloudesireClient( ObjectMapper mapper, String baseUrl, String userAgent, Timeout timeout )
     {
         this.baseUrl = baseUrl;
         this.mapper = mapper;
 
         okhttpClientBuilder = new OkHttpClient.Builder()
-                .connectTimeout( 30, TimeUnit.SECONDS )     // connect timeout
-                .readTimeout( 60, TimeUnit.SECONDS );       // socket timeout
+                .connectTimeout( timeout.getConnectTimeout(), TimeUnit.MILLISECONDS )     // connect timeout
+                .readTimeout( timeout.getReadTimeout(), TimeUnit.MILLISECONDS );          // socket timeout
```

Comment:
take a look at the alternative methods `#connectTimeout(Duration duration)` and `#readTimeout(Duration duration)`

Reply:
Duration is available since java 8 base client is java 7 :cry: